src/reporter.ts

```diff
@@ -0,0 +1,253 @@
+    this.cucumberVersion = 'unknown'
+    this.testRun = new TestRun();
+    this.assets = [];
+    this.consoleLog = [];
+    this.passed = true;
+    this.startedAt = new Date().toISOString();
+
+    let reporterVersion = 'unknown';
+    try {
+      const packageData = JSON.parse(fs.readFileSync(path.join(__dirname, '..', 'package.json'), 'utf-8'));
+      reporterVersion = packageData.version;
+    // eslint-disable-next-line no-empty
+    } catch (e) {}
+
+    if (process.env.SAUCE_USERNAME && process.env.SAUCE_USERNAME !== '' && process.env.SAUCE_ACCESS_KEY && process.env.SAUCE_ACCESS_KEY !== '') {
+      this.api = new SauceLabs({
+        user: process.env.SAUCE_USERNAME,
+        key: process.env.SAUCE_ACCESS_KEY,
+        region: this.region,
+        headers: {
+          'User-Agent': `cucumber-reporter/${reporterVersion}`
+        },
+      });
+      this.api.tld = this.region === 'staging' ? 'net' : 'com';
+    }
+    if (process.env.SAUCE_VIDEO_START_TIME) {
+      this.videoStartTime = new Date(process.env.SAUCE_VIDEO_START_TIME).getTime();
+    }
+
+    config.eventBroadcaster?.on('envelope', async (envelope: Envelope) => {
+      if (envelope.testCaseFinished) {
+        this.logTestCase(envelope.testCaseFinished)
+      }
+      if (envelope.testRunFinished) {
+        this.endedAt = new Date().toISOString();
+        await this.logTestRun(envelope.testRunFinished)
+      }
+    })
+  }
+
+  logTestCase(testCaseFinished: { testCaseStartedId: any }) {
+    const testCaseAttempt = this.eventDataCollector.getTestCaseAttempt(testCaseFinished.testCaseStartedId)
+    const parsed = formatterHelpers.parseTestCaseAttempt({
+      snippetBuilder: this.snippetBuilder, 
+      supportCodeLibrary: this.supportCodeLibrary,
+      testCaseAttempt 
+    })
+    const suite = new Suite(parsed.testCase?.sourceLocation?.uri || '')
+    const curr = new Suite(parsed.testCase?.name);
+    curr.metadata = {
+      attempt: parsed.testCase?.attempt,
+      sourceLocation: parsed.testCase?.sourceLocation,
+    }
+
+    parsed.testSteps.forEach((testStep) => {
+      const test = new Test(`${testStep.keyword}${testStep.text || ''}`)
+      const testStatus = testStep.result?.status?.toLowerCase();
+      test.status = testStatus === 'skipped' ? SauceStatus.Skipped : (testStatus === 'passed' ? SauceStatus.Passed : SauceStatus.Failed);
+      test.output = testStep.result?.message;
+      test.duration = this.durationToMilliseconds(testStep.result?.duration);
+      test.attachments = [];
+      testStep.attachments.forEach((attachment: any) => {
+        this.assets.push({
+          filename: attachment.testCaseStartedId + '.log',
+          data: Buffer.from(attachment.body),
+        })
+        test.attachments?.push({
+          name: attachment.testCaseStartedId + '.log',
+          contentType: attachment.mediaType,
+          path: ''
+        })
+      })
+      curr.addTest(test);
+    })
+    curr.status = this.testRun.computeStatus()
```

Comment:
You should computeStatus after all the suites have been added. Otherwise you're skipping the current suite when computing the current status.

Reply:
Here's the implementation for reference: https://github.com/saucelabs/sauce-json-reporter-js/blob/main/src/index.ts#L45-L52